File: 349-intersection-of-two-arrays/intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        map<int, int> mp;

        for(auto it : nums1){
            mp[it] = 1;
        }

        vector<int> ans;

        for(auto it : nums2){
            if(mp[it] == 1) ans.push_back(it), mp[it] = 0;
        }

        return ans;
    }
};

File: 349-intersection-of-two-arrays/intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        boolean[] arr1 = new boolean[1001];
        boolean[] arr2 = new boolean[1001];
        
        for (int num : nums1) {
            arr1[num] = true;
        }
        for (int num : nums2) {
            arr2[num] = true;
        }
        List<Integer> list = new ArrayList<>();
        for (int i = 0; i <= 1000; i++) {
            if (arr1[i] && arr2[i]) {
                list.add(i);
            }
        }
        int[] ans = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            ans[i] = list.get(i);
        }
    
        return ans;
    }
}
